src/lib/utils/fetch-user-history.ts

```diff
@@ -0,0 +1,61 @@
+import 'server-only';
+import { dbAdmin } from '@/lib/firebase-admin';
+import { type HttpHeader, type RequestRecord, isMethod } from '@/lib/types';
+
+export async function fetchUserHistory(
+  uid: string,
+  limit = 100
+): Promise<RequestRecord[]> {
+  const snap = await dbAdmin
+    .collection('users')
+    .doc(uid)
+    .collection('requests')
+    .orderBy('time', 'desc')
+    .limit(limit)
+    .get();
+
+  const rows: RequestRecord[] = snap.docs.map(doc => {
+    const data = doc.data();
+
+    const restoreRaw =
+      data && typeof data === 'object' && data.restore ? data.restore : {};
+    const headers: HttpHeader[] =
+      restoreRaw &&
+      typeof restoreRaw.headers === 'object' &&
+      !Array.isArray(restoreRaw.headers)
+        ? Object.entries(restoreRaw.headers).map(([key, value], index) => ({
+            id: `${doc.id}-${index}`,
+            key: String(key),
+            value: String(value),
+          }))
+        : [];
+
+    const method = isMethod(data.method) ? data.method : 'GET';
```

Comment:
DEFAULT_HTTP_METHOD лежит в константах

```suggestion
    const method = isMethod(data.method) ? data.method : DEFAULT_HTTP_METHOD
```